### PR TITLE
[S3C-3811] Add a docker-build stage in the CI, updated ReadMe

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**

### CODE REVIEW

**Review by anurag4DSB — COMMENTED** [review]
I tried the instructions and got this:

```
❯ docker run --env-file /Users/anurag/Projects/scality/vmware-ose-scality/src/main/resources/application.properties -it -p8443:8443 osis-scality-0.1.0-snapshot
Unable to find image 'osis-scality-0.1.0-snapshot:latest' locally
docker: Error response from daemon: pull access denied for osis-scality-0.1.0-snapshot, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
```

Do we need to use a specific docker image tag, rather than the latest?

File: README.md
Comment (anurag4DSB) [inline README.md:28]: Can we also make the version generic here?
As I tried these commands it produced `osis-scality-0.0.1-SNAPSHOT.jar`

File: README.md
Comment (anurag4DSB) [inline README.md:25]: What do we need to update?

File: eve/main.yml
Comment (anurag4DSB) [inline eve/main.yml:104]: New line in this and other files.

File: src/main/resources/application.yml
Comment (anurag4DSB) [inline src/main/resources/application.yml:40]: Do we still need this file?

File: Dockerfile
Comment (anurag4DSB) [inline Dockerfile:10]: Same new line.

File: eve/workers/docker/Dockerfile
Comment (anurag4DSB) [inline eve/workers/docker/Dockerfile:25]: Here too new line needed.

File: src/main/resources/application.properties
Comment (anurag4DSB) [inline src/main/resources/application.properties:48]: new line here as well.

File: src/main/resources/application.yml
Comment (anurag4DSB) [reply src/main/resources/application.yml:40]: and new line

File: eve/main.yml
Comment (tcarmet) [inline eve/main.yml:82]: Question for you, so here you are doing the following pipeline:
```
└── test # I test my component and only when I finished testing it
    └── docker-build # I start building the image
```
Where you could do:
```
├── test # I test my component
└── docker-build # At the same time I build my docker image
```

Do you want to continue with the first option? or do you want me to suggest how you could move to option 2?

Comment (saitharunthodupunuru) [reply eve/main.yml:82]: @tcarmet here I want to build and push the docker image only after I do the testing. If my tests fail then no docker image. Please let me know if anything we can do parallel in this approach

File: src/main/resources/application.yml
Comment (saitharunthodupunuru) [reply src/main/resources/application.yml:40]: removing this

File: eve/main.yml
Comment (tcarmet) [reply eve/main.yml:82]: On the `gradle build` step, add the following option `haltOnFailure: true`. So that the pipeline will achieve what you desire.

File: README.md
Comment (saitharunthodupunuru) [reply README.md]: Updated to generic version

**Comment by saitharunthodupunuru** [comment]
> I tried the instructions and got this:
> 
> ```
> ❯ docker run --env-file /Users/anurag/Projects/scality/vmware-ose-scality/src/main/resources/application.properties -it -p8443:8443 osis-scality-0.1.0-snapshot
> Unable to find image 'osis-scality-0.1.0-snapshot:latest' locally
> docker: Error response from daemon: pull access denied for osis-scality-0.1.0-snapshot, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
> See 'docker run --help'.
> ```
> 
> Do we need to use a specific docker image tag, rather than the latest?

Updated the main.yml to add the tag for "latest" so docker pull and run can be done with "latest" tag

File: eve/main.yml
Comment (saitharunthodupunuru) [reply eve/main.yml:83]: Done. Thank you @tcarmet

File: README.md
Comment (anurag4DSB) [inline README.md:17]: needs to be a coded block.

File: README.md
Comment (anurag4DSB) [inline README.md:24]: same here, it needs to be a code block.

File: README.md
Comment (anurag4DSB) [inline README.md:29]: you can use something like this

```sh
$ docker pull registry.scality.com/vmware-ose-scality/vmware-ose-scality:<tag>
$ docker run --env-file application.properties -it -p8443:8443 registry.scality.com/vmware-ose-scality/vmware-ose-scality:<tag>
```

**Review by anurag4DSB — COMMENTED** [review]
The commands need to be in a code block. ANd this project eventually needs a markdown linter.

**Comment by saitharunthodupunuru** [comment]
@anurag4DSB  Please recheck my latest commit https://github.com/scality/vmware-ose-scality/commit/84d88ba7a955962b9af08746e28f5a889cf829ae as suggested.

@tcarmet  Waiting for your approval. Changed as suggested

**Review by anurag4DSB — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

**Review by tcarmet — APPROVED** [review]

**Pull request merged**
